remote/analyze_handler.py

```diff
@@ -0,0 +1,164 @@
+
+    build_command_for_codechecker = build_command.replace(read_in_file_path, analyze_dir_path + '/sources-root' + read_in_file_path)
+
+    codechecker_path = os.path.join(os.getcwd(), "build/CodeChecker/bin/CodeChecker")
+
+    command = [codechecker_path, "check"]
+    command.append("-b")
+    command.append("%s" % build_command_for_codechecker)
+    command.append("-o")
+    command.append("%s" % os.path.join(analyze_dir_path, 'output'))
+
+    LOGGER.debug("CodeChecker command: %s" % command)
+
+    REDIS_DATABASE.hset(analyze_id, 'state',
+                            AnalyzeStatus.ANALYZE_IN_PROGRESS.name)
+
+    process = subprocess.Popen(command,
+                               stdin=subprocess.PIPE,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE)
+
+    stdout, stderr = process.communicate()
+    returncode = process.wait()
+
+    LOGGER.debug("Command output: \n%s" % stdout)
+
+    with open("stdout", "w") as text_file:
+        text_file.write(stdout)
+
+    LOGGER.debug("Error output: %s" % stderr)
+
+    with open("stderr", "w") as text_file:
+        text_file.write(stderr)
+
+    LOGGER.debug("Return code: %s" % returncode)
+
+    with open("returncode", "w") as text_file:
+        text_file.write(str(returncode))
+
+
+def postAnalyze(analyze_id, part_number):
+    LOGGER.info('Post Analyze Step')
+
+    file_name = 'source'
+    file_extension = '.zip'
+
+    file_path = os.path.join('/workspace', analyze_id)
+
+    source_file_without_extension = file_name + '_' + str(part_number)
+
+    analyze_dir_path = os.path.join(file_path, source_file_without_extension)
+
+    output_dir_path = os.path.join(analyze_dir_path, 'output')
+
+    analyze_output_zip_file_path = os.path.join(file_path, 'output.zip')
+
+    with zipfile.ZipFile(analyze_output_zip_file_path, 'a') as output:
+        for root, dirs, files in os.walk(output_dir_path):
+            for f in files:
+                output.write(os.path.join(root, f))
+
+    REDIS_DATABASE.hincrby(analyze_id, 'completed_parts', 1)
+
+    parts = REDIS_DATABASE.hget(analyze_id, 'parts')
+    completed_parts = REDIS_DATABASE.hget(analyze_id, 'completed_parts')
+
+    if parts == completed_parts:
+        REDIS_DATABASE.hset(analyze_id, 'state',
+                            AnalyzeStatus.ANALYZE_COMPLETED.name)
+
+
+def main():
+    scheduler = sched.scheduler(time.time, time.sleep)
+
+    def check_queue():
+        scheduler.enter(10, 1, check_queue, ())
+        LOGGER.info("Check task queue in Redis.")
+        tasks = REDIS_DATABASE.lrange('ANALYSES_QUEUE', 0, -1)
+
+        task = REDIS_DATABASE.lpop('ANALYSES_QUEUE')
+
+        if task is not None:
+            LOGGER.info("Got a task: %s, starting analyze it, leftover tasks %s" % (task, len(tasks)))
+            analyze_id, part_number = str(task).split('-')
+            preAnalyze(analyze_id, part_number)
```

Comment:
I would handle every exception in `preAnalyze`, and `postAnalyze` so they are less likely to crash, however docker should restart the container in case of an error.

Reply:
Done